cmd/erasure-server-pool.go

```diff
@@ -1713,8 +1785,16 @@ func (z *erasureServerPools) ListMultipartUploads(ctx context.Context, bucket, p
 		if err != nil {
 			return result, err
 		}
-		poolResult.Uploads = append(poolResult.Uploads, result.Uploads...)
+		// poolResult.Uploads = append(poolResult.Uploads, result.Uploads...)
+		uploads = append(uploads, result.Uploads...)
+	}
+
+	for _, upload := range uploads {
+		if _, found := z.multipartUploadsCache.Get(upload.UploadID); found {
```

Comment:
```suggestion
		if ok := z.multipartUploadsCache.Has(upload.UploadID); ok {
```

implement a Has(key) method to make this simpler